### ISSUE TITLE
how to select elements in a tensor by 2d index?

### ISSUE THREAD

**Issue by danliu2** [pytorch/pytorch#1651]
Hi! I need to select elements by 2d index, just like in numpy:
label =[10,15,20]
probs= numpy.random.rand(3,40)
target_probs = probs[numpy.arange(probs.shape[0]), label]

this function is needed while I'm doing reinforcement learning, and I want to log the expected rewards while training.

**Comment by apaszke** [comment]
You can try with `torch.gather`.

Note that we're using GitHub issues for bug reports only, and all questions should be posted on [our forums](discuss.pytorch.org).
